cli/src/cast.rs

```diff
@@ -710,6 +728,41 @@ where
     })
 }
 
+fn det_base_in(value: &str, base_in: Option<String>) -> eyre::Result<u32> {
+    match base_in {
+        Some(base_in) => match base_in.as_str() {
+            "10" | "dec" => Ok(10),
+            "16" | "hex" => Ok(16),
+            _ => eyre::bail!("Unknown input base: {base_in}"),
+        },
+        None if value.starts_with("0x") => Ok(16),
+        None => match U256::from_str_radix(value, 10) {
+            Ok(_) => {
+                eyre::bail!("Could not autodetect input base: input could be decimal or hexadecimal. Please prepend with 0x if the input is hexadecimal, or specify a --base-in parameter.");
+            }
+            Err(_) => {
+                U256::from_str_radix(value, 16).expect("Could not autodetect input base.");
+                Ok(16)
+            }
+        },
+    }
+}
+
+fn det_base_out(base_out: &str) -> u32 {
+    match base_out {
+        "10" | "dec" => 10,
+        "16" | "hex" | _ => 16,
```

Comment:
this will match anything due to the `_` wildcard pattern, is this intended here?

Reply:
Right, so the aim was to default the base out to 16 but we might prefere to eyre::bail! a message for the user here

Reply:
I'd throw an error to the user in this case